code/modules/mob/living/carbon/xenomorph/castes/king/abilities_conqueror.dm

```diff
@@ -0,0 +1,979 @@
+				continue
+			var/mob/living/living_target = mob_target
+			targets_to_attack += living_target
+			RegisterSignals(living_target, list(COMSIG_QDELETING, COMSIG_MOB_DEATH, COMSIG_MOVABLE_Z_CHANGED), PROC_REF(clear_ref))
+	if(!length(targets_to_attack))
+		disable_ability()
+		return
+	if(xeno_owner.buckled)
+		xeno_owner.buckled.unbuckle_mob(xeno_owner, TRUE)
+	xeno_owner.alpha = 0
+	xeno_owner.set_canmove(FALSE)
+	xeno_owner.status_flags |= (GODMODE|INCORPOREAL)
+	playsound(xeno_owner, 'sound/effects/alien/conqueror/obliteration_roar.ogg', 40, TRUE, 15)
+	attack_targets()
+
+/// Removes a target from the list and clears related signals.
+/datum/action/ability/xeno_action/conqueror_obliteration/proc/clear_ref(datum/source)
+	SIGNAL_HANDLER
+	targets_to_attack -= source
+	UnregisterSignal(source, list(COMSIG_QDELETING, COMSIG_MOB_DEATH, COMSIG_MOVABLE_Z_CHANGED))
+
+/// Attacks all eligible targets, inflicting damage and creating visuals.
+/datum/action/ability/xeno_action/conqueror_obliteration/proc/attack_targets()
+	var/mob/living/living_target = targets_to_attack[1]
+	var/turf/current_turf = xeno_owner.loc
+	var/turf/new_turf = get_step_rand(living_target.loc)
+	for(var/x = 1 to get_dist(current_turf, new_turf))
+		var/turf_to_check = get_step(current_turf, get_dir(current_turf, new_turf))
+		if(turf_to_check && !DirBlocked(current_turf, turf_to_check, xeno_owner.pass_flags))
+			new /obj/effect/temp_visual/conqueror/dash_trail(turf_to_check, get_dir(current_turf, turf_to_check))
+		current_turf = turf_to_check
+	playsound(new_turf, 'sound/effects/alien/behemoth/landslide_enhanced_charge.ogg', 8, TRUE)
+	playsound(living_target, 'sound/effects/alien/conqueror/will_kick.ogg', 40, TRUE)
+	new /obj/effect/temp_visual/conqueror/hook/punch(living_target.loc)
+	xeno_owner.forceMove(new_turf)
+	living_target.do_jitter_animation(700, CONQUEROR_OBLITERATION_DEBUFF)
+	living_target.Immobilize(CONQUEROR_OBLITERATION_DEBUFF SECONDS)
+	living_target.adjust_stagger(CONQUEROR_OBLITERATION_DEBUFF)
+	living_target.adjust_slowdown(CONQUEROR_OBLITERATION_DEBUFF)
+	living_target.apply_damage((xeno_owner.xeno_caste.melee_damage * xeno_owner.xeno_melee_damage_modifier) * CONQUEROR_OBLITERATION_DAMAGE_MULTIPLIER, BRUTE, 0, MELEE, TRUE, TRUE, TRUE, xeno_owner.xeno_caste.melee_ap)
+	INVOKE_ASYNC(living_target, TYPE_PROC_REF(/mob, emote), "gored")
+	UnregisterSignal(living_target, list(COMSIG_QDELETING, COMSIG_MOB_DEATH, COMSIG_MOVABLE_Z_CHANGED))
+	targets_to_attack -= living_target
+	if(length(targets_to_attack))
+		attack_timer = addtimer(CALLBACK(src, PROC_REF(attack_targets)), CONQUEROR_OBLITERATION_ATTACK_DELAY, TIMER_UNIQUE|TIMER_OVERRIDE|TIMER_STOPPABLE)
+		return
+	disable_ability()
+
+/// Disables the ability.
+/datum/action/ability/xeno_action/conqueror_obliteration/proc/disable_ability()
```

Comment:
less a disable and more an end tbh? this looks like it just cleans up when we're done TPIng

Reply:
Updated the proc name and the autodoc in the next commit.